### ISSUE TITLE
chromium: vulkan is not an allowed implementation

### ISSUE THREAD

**Issue by tlvince** [NixOS/nixpkgs#320030]
### Describe the bug

vaapi on amd cards [requires `--use-angle=vulkan`](https://wiki.archlinux.org/title/Chromium#Hardware_video_acceleration).

### Steps To Reproduce

In wayland, with `NIXOS_OZONE_WL=1`:

```
❯ chromium --use-angle=vulkan
[6893:6893:0615/112355.605938:ERROR:gl_factory.cc(120)] Requested GL implementation (gl=egl-angle,angle=vulkan) not found in allowed implementations: [(gl=egl-angle,angle=opengl),(gl=egl-angle,angle=opengles),(gl=egl-gles2,angle=none),(gl=egl-angle,angle=swiftshader)].
```

### Expected behavior

vulkan to be allowed

### Additional context

Per [upstream issue](https://issues.chromium.org/issues/334275637), a [one-liner patch](https://chromium.googlesource.com/chromium/src/+/9ca21ac45af570ce11a1b0b96fdb163985b59178%5E%21/#F0) is needed to enable it.

Arch Linux [recently added it](https://gitlab.archlinux.org/archlinux/packaging/packages/chromium/-/merge_requests/6).

Now that #306791 is in nixos-unstable, now's a good time to enable this.

Note, upstream did [revert the change](https://chromium-review.googlesource.com/c/chromium/src/+/5568860) - there's probably some edge case, _but_ it requires `--use-angle=vulkan` to be explicitly passed and AFAIK, there are no bug reports from Arch Linux users. It's very useful for AMD users.

### Notify maintainers

@emilylange, @networkexception, @Kiskae 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
> there are no bug reports from Arch Linux users

To be fair, it was only added 2 weeks ago, so that might not really be a huge data point either way.
https://gitlab.archlinux.org/archlinux/packaging/packages/chromium/-/commit/07e8575a6f2151055d480a7578687704c72019f2

**Comment by tlvince** [comment]
That's fair, though if I'm following the [upstream issue](https://chromium-review.googlesource.com/c/chromium/src/+/5568860/3#message-d16a4ab8c83863a64d94b89b7f17df93b0de67bf) correctly, it was reverted due to failing [Lacros tests](https://issuetracker.google.com/issues/343754367) (Chrome OS-related).

Given the patch is a no-op unless the Vulkan CLI flag is passed, it'd be nice to have or some other way to test.

**Comment by emilylange** [comment]
I am not particularly sold on adding that patch.

It looks like it could be part of M127 (if the revert is reverted again).
And M127 is scheduled for 2024-07-23, so not too far away (~5 weeks).

So I would rather wait for M127 instead.

**Comment by tlvince** [comment]
No probs! I'll close this then.

**Comment by k1gen** [comment]
waiting on 129: https://chromium-review.googlesource.com/c/chromium/src/+/5740783
